### PR TITLE
Make `cirq_gate_as_bloq` public

### PR DESCRIPTION
**Pull request by mpharrigan**
This is generally useful when switching between Cirq gates and bloqs that are basic gates; part of #1109

### CODE REVIEW

File: qualtran/cirq_interop/_cirq_to_bloq.py [hunk 1]
```diff
@@ -378,6 +378,11 @@ def _cirq_gate_to_bloq(gate: cirq.Gate) -> Bloq:
     if gate in CIRQ_GATE_TO_BLOQ_MAP:
         return CIRQ_GATE_TO_BLOQ_MAP[gate]
 
+    if isinstance(gate, cirq.ControlledGate):
+        return Controlled(
+            cirq_gate_to_bloq(gate.sub_gate), CtrlSpec.from_cirq_cv(gate.control_values)
+        )
+
```
Comment (tanujkhattar) [inline qualtran/cirq_interop/_cirq_to_bloq.py:385]: Why this change?

Comment (mpharrigan) [reply qualtran/cirq_interop/_cirq_to_bloq.py:385]: for #1115  or any case where we have an explicit bloq that takes priority over the general `Controlled` wrap.

**Comment by mpharrigan** [comment]
(the other basic gates were based on top of this change, so merging this will cause those tests to pass)

**Review by tanujkhattar — APPROVED** [review]
